internal/services/extendedlocation/extended_location_custom_location_test.go

```diff
@@ -0,0 +1,352 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package extendedlocation_test
+
+import (
+	"context"
+	cryptoRand "crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"math/rand"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/extendedlocation/2021-08-15/customlocations"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type CustomLocationResource struct{}
+
+func (r CustomLocationResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := customlocations.ParseCustomLocationID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.ExtendedLocation.CustomLocationsClient.Get(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+	return utils.Bool(true), nil
+}
+
+func TestAccExtendedLocationCustomLocations_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_extended_custom_location", "test")
+	r := CustomLocationResource{}
+	credential, privateKey, publicKey := r.getCredentials(t)
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data, credential, privateKey, publicKey),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccExtendedLocationCustomLocations_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_extended_custom_location", "test")
+	r := CustomLocationResource{}
+	credential, privateKey, publicKey := r.getCredentials(t)
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data, credential, privateKey, publicKey),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data, credential, privateKey, publicKey),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
```

Comment:
Could we add one more testcase to test whether we could deploy the complete config directly instead of updating it from the basic config? And consider updating the config name from "update" to "complete"?

Reply:
Will fix.